pkg/hakeeper/checkers/coordinator.go

```diff
@@ -15,32 +15,57 @@
 package checkers
 
 import (
+	"sync"
+
 	"github.com/matrixorigin/matrixone/pkg/hakeeper/checkers/dnservice"
 	"github.com/matrixorigin/matrixone/pkg/hakeeper/checkers/logservice"
+	"github.com/matrixorigin/matrixone/pkg/hakeeper/checkers/syshealth"
 	"github.com/matrixorigin/matrixone/pkg/hakeeper/checkers/util"
 	"github.com/matrixorigin/matrixone/pkg/hakeeper/operator"
 	hapb "github.com/matrixorigin/matrixone/pkg/pb/hakeeper"
 )
 
 type Coordinator struct {
 	OperatorController *operator.Controller
+
+	mu struct {
```

Comment:
I wonder why we need lock here. Everything in hakeeper is single-threaded.

Reply:
I've no idea whether it's single-threaded or not.
Even it's single-threaded, a mutex could make the update for `teardown` and `operators` as atomic.
Just think program written in python which is usually with GIL.